src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -0,0 +1,1106 @@
+                    in the shifted basis
+                    sage: A.algebra_generators()
+                    Family (S[1],)
+                """
+                NonNeg = self.basis().keys()
+                return Family([self.monomial(NonNeg(1))])
+
+            gens = algebra_generators
+
+        class ElementMethods:
+            def __call__(self, v):
+                """
+                Evaluation at some value ``v``
+
+                EXAMPLES::
+
+                     sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                     sage: B = F.gen()
+                     sage: f = B**2+4*B+6
+                     sage: f(1/3)
+                     118/9
+
+                     sage: F = IntegerValuedPolynomialRing(ZZ).B()
+                     sage: B = F.gen()
+                     sage: f = B**2+4*B+6
+                     sage: f(1/3)
+                     67/9
+                """
+                return self.polynomial()(v)
+
+            def polynomial(self):
+                """
+                Convert to a polynomial in `x`.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.gen()
+                    sage: (B+1).polynomial()
+                    x + 2
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).B()
+                    sage: B = F.gen()
+                    sage: (B+1).polynomial()
+                    x + 1
+
+                TESTS::
+
+                    sage: F.zero().polynomial().parent()
+                    Univariate Polynomial Ring in x over Rational Field
+                """
+                R = PolynomialRing(QQ, 'x')
+                p = self.parent()._poly
+                return R.sum(c * p(i) for i, c in self)
+
+            def shift(self, j=1):
+                """
+                Shift all indices by `j`.
+
+                INPUT:
+
+                - `j` -- integer (default: 1)
+
+                In the binomial basis, the shift by 1 corresponds to
+                a summation operator from `0` to `x`.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).B()
+                    sage: B = F.gen()
+                    sage: (B+1).shift()
+                    B[1] + B[2]
+                    sage: (B+1).shift(3)
+                    B[3] + B[4]
+                """
+                A = self.parent()
+                return A._from_dict({A._indices(i + j): c for i, c in self})
+
+            def sum_of_coefficients(self):
+                """
+                Return the sum of coefficients.
+
+                In the shifted basis, this is the evaluation at `x=0`.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.basis()
+                    sage: (B[2]*B[4]).sum_of_coefficients()
+                    1
+                """
+                return sum(c for _, c in self)
```

Comment:
```suggestion
                return self.parent().base_ring().sum(self._monomial_coefficients.values())
```
The first part is needed for `0`, which would otherwise be an `int`.